File: internal/screens/event_overview.go
package screens

import (
	"fmt"
	"strings"

	"github.com/jashkarangiya/valo-tui/internal/data"
	"github.com/jashkarangiya/valo-tui/internal/styles"
)

// EventOverview is the [o] event landing page: what the event is and where
// things stand, before drilling into the children.
type EventOverview struct {
	w, h int
	body string
}

func NewEventOverview(w, h int) EventOverview { return EventOverview{w: w, h: h} }

func (s *EventOverview) SetSize(w, h int) { s.w, s.h = w, h }

func (s *EventOverview) Load(eventID int, eventName string) {
	var b strings.Builder
	b.WriteString(title("overview") + "\n\n")
	if eventID == 0 {
		b.WriteString(muted("select an event from ") + text("e events"))
		s.body = b.String()
		return
	}
	event, _ := data.EventByID(eventID)
	name := event.Name
	if name == "" {
		name = eventName
	}
	if name == "" {
		name = "event"
	}
	matches := data.EventMatchCards(eventID, name)

	b.WriteString(s.banner(event, name) + "\n\n")
	b.WriteString(s.progress(matches) + "\n")
	b.WriteString(s.nav())
	s.body = b.String()
}

func (s EventOverview) banner(e data.EventCard, name string) string {
	region := e.Region
	if region == "" {
		region = "—"
	}
	status := e.Status
	if status == "" {
		status = "—"
	}
	statusTxt := text(status)
	if strings.HasPrefix(strings.ToLower(status), "ongo") {
		statusTxt = live(status)
	}
	lines := []string{
		textB(name),
		muted("region ") + text(region) + "   " + muted("status ") + statusTxt,
	}
	if e.Start != "" && e.End != "" {
		lines = append(lines, muted("dates  ")+text(e.Start+" – "+e.End))
	} else if e.Start != "" {
		lines = append(lines, muted("dates  ")+text(e.Start))
	}
	if e.Prize != "" {
		lines = append(lines, muted("prize  ")+text(e.Prize))
	}
	return styles.Card.Render(strings.Join(lines, "\n"))
}

func (s EventOverview) progress(matches []data.MatchCard) string {
	var liveN, done, soon int
	var phases []string
	seen := map[string]bool{}
	for _, m := range matches {
		switch m.Status {
		case "live":
			liveN++
		case "completed":
			done++
		case "upcoming":
			soon++
		}
		if m.Phase != "" && !seen[m.Phase] {
			seen[m.Phase] = true
			phases = append(phases, m.Phase)
		}
	}

	var b strings.Builder
	b.WriteString(accentB("progress") + "\n")
	b.WriteString(muted("matches  ") + text(fmt.Sprintf("%d done", done)) + muted(" · ") +
		live(fmt.Sprintf("%d live", liveN)) + muted(" · ") + text(fmt.Sprintf("%d upcoming", soon)))
	if len(phases) == 0 {
		b.WriteString("\n" + muted("no matches cached for this event yet"))
		return b.String()
	}
	b.WriteString("\n\n")
	for i, ph := range phases {
		if i >= 8 {
			break
		}
		var left, liveHere int
		for _, m := range matches {
			if m.Phase != ph {
				continue
			}
			if m.Status != "completed" {
				left++
			}
			if m.Status == "live" {
				liveHere++
			}
		}
		state := text(fmt.Sprintf("%d left", left))
		if liveHere > 0 {
			state = live("live")
		} else if left == 0 {
			state = muted("complete")
		}
		b.WriteString(text(fmt.Sprintf("%-22s", ph)) + " " + state + "\n")
	}
	return strings.TrimRight(b.String(), "\n")
}

func (s EventOverview) nav() string {
	return accentB("quick nav") + "\n" +
		muted("· ") + text("r") + muted("  results    completed & live series") + "\n" +
		muted("· ") + text("f") + muted("  fixtures   what's still to come") + "\n" +
		muted("· ") + text("t") + muted("  standings  group tables") + "\n" +
		muted("· ") + text("b") + muted("  bracket    playoff tree") + "\n" +
		muted("· ") + text("m") + muted("  teams      rosters in this event")
}

func (s EventOverview) View() string { return s.body }

File: valo_tui/screens/event_overview.py
"""[o] overview — the event landing page.

The parent of all the event sub-pages: it tells you what the event *is*
(region, format, status, dates) and where things stand (phases seen, how many
matches are done / live / upcoming) before you drill into the children.
"""

from __future__ import annotations

from textual.app import ComposeResult
from textual.containers import VerticalScroll
from textual.widgets import Label, Static

from ..data import cache
from ..data.models import MatchCard
from .widgets import ACCENT, LIVE, MUTED, TEXT


class EventOverviewView(VerticalScroll):
    can_focus = True

    def compose(self) -> ComposeResult:
        yield Label("overview", classes="page-title")
        yield Static(id="ov-banner", classes="card")
        yield Static(id="ov-progress")
        yield Static(id="ov-nav")

    def load_data(self) -> None:
        eid = getattr(self.app, "current_event_id", None)
        if eid is None:
            self.query_one("#ov-banner", Static).update(
                f"[{MUTED}]select an event from [/][{TEXT}]e events[/]"
            )
            self.query_one("#ov-progress", Static).update("")
            self.query_one("#ov-nav", Static).update("")
            return

        event = cache.event_by_id(eid)
        name = event.name if event else "event"
        matches = cache.event_match_cards(eid, name)
        self._render_banner(event, name)
        self._render_progress(matches)
        self._render_nav()

    def _render_banner(self, event, name: str) -> None:
        region = (event.region if event else None) or "—"
        status = (event.status if event else None) or "—"
        dates = ""
        if event and event.start and event.end:
            dates = f"{event.start} – {event.end}"
        elif event and event.start:
            dates = event.start
        prize = (event.prize if event else None) or ""

        lines = [f"[bold {TEXT}]{name}[/]"]
        meta = f"[{MUTED}]region [/][{TEXT}]{region}[/]   [{MUTED}]status [/]"
        meta += f"[{LIVE}]{status}[/]" if status.lower().startswith("ongo") else f"[{TEXT}]{status}[/]"
        lines.append(meta)
        if dates:
            lines.append(f"[{MUTED}]dates  [/][{TEXT}]{dates}[/]")
        if prize:
            lines.append(f"[{MUTED}]prize  [/][{TEXT}]{prize}[/]")
        self.query_one("#ov-banner", Static).update("\n".join(lines))

    def _render_progress(self, matches: list[MatchCard]) -> None:
        live = sum(1 for m in matches if m.status == "live")
        done = sum(1 for m in matches if m.status == "completed")
        soon = sum(1 for m in matches if m.status == "upcoming")

        # Distinct phases, in the order they first appear, with their completion.
        phases: list[str] = []
        for m in matches:
            if m.phase and m.phase not in phases:
                phases.append(m.phase)

        lines = [
            f"[bold {ACCENT}]progress[/]",
            f"[{MUTED}]matches  [/][{TEXT}]{done} done[/][{MUTED}] · [/]"
            f"[{LIVE}]{live} live[/][{MUTED}] · [/][{TEXT}]{soon} upcoming[/]",
        ]
        if phases:
            lines.append("")
            for ph in phases[:8]:
                in_phase = [m for m in matches if m.phase == ph]
                left = sum(1 for m in in_phase if m.status != "completed")
                if any(m.status == "live" for m in in_phase):
                    state = f"[{LIVE}]live[/]"
                elif left == 0:
                    state = f"[{MUTED}]complete[/]"
                else:
                    state = f"[{TEXT}]{left} left[/]"
                lines.append(f"[{TEXT}]{ph:<22}[/] {state}")
        else:
            lines.append(f"[{MUTED}]no matches cached for this event yet[/]")
        self.query_one("#ov-progress", Static).update("\n".join(lines))

    def _render_nav(self) -> None:
        nav = (
            f"\n[bold {ACCENT}]quick nav[/]\n"
            f"[{MUTED}]· [/][{TEXT}]r[/][{MUTED}]  results    completed & live series[/]\n"
            f"[{MUTED}]· [/][{TEXT}]f[/][{MUTED}]  fixtures   what's still to come[/]\n"
            f"[{MUTED}]· [/][{TEXT}]t[/][{MUTED}]  standings  group tables[/]\n"
            f"[{MUTED}]· [/][{TEXT}]b[/][{MUTED}]  bracket    playoff tree[/]\n"
            f"[{MUTED}]· [/][{TEXT}]m[/][{MUTED}]  teams      rosters in this event[/]"
        )
        self.query_one("#ov-nav", Static).update(nav)
